iothub_client/tests/iothubtransporthttp_ut/iothubtransporthttp_ut.c

```diff
@@ -3532,12 +3554,15 @@ TEST_FUNCTION(IoTHubTransportHttp_DoWork_happy_path_with_2_registered_devices_em
 //Tests_SRS_TRANSPORTMULTITHTTP_17_100: [ Accepting a message is successful when HTTPAPIEX_SAS_ExecuteRequest completes successfully and the status code is 204. ]
 //Tests_SRS_TRANSPORTMULTITHTTP_17_090: [ All the HTTP headers of the form iothub-app-name:somecontent shall be transformed in message properties {name, somecontent}. ]
 //Tests_SRS_TRANSPORTMULTITHTTP_17_123: [ After client creation, the first GET shall be allowed no matter what the value of GetMinimumPollingTime. ]
-TEST_FUNCTION(IoTHubTransportHttp_DoWork_happy_path_with_empty_waitingToSend_and_1_service_message_succeeds)
+TEST_FUNCTION(IoTHubTransportHttp_DoWork_happy_path_with_empty_waitingToSend_and_1_service_message_fail_set_disposition_fails)
 {
     //arrange
     unsigned int statusCode200 = 200;
     unsigned int statusCode204 = 204;
 
+    char* real_ETAG = (char*)malloc(sizeof(TEST_ETAG_VALUE) + 1);
```

Comment:
nit: check for `NULL`?

Reply:
On tests we are more lax, since they are usually run on higher capacity platforms and malloc failures would be really unexpected.